metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -0,0 +1,1152 @@
+                params.target.toString(),
+              )
+            Future.successful(()).asJavaObject
+        }
+      case ServerCommands.ResolveAndStartTestSuite(params)
+          if params.testClass != null =>
+        Future
+          .sequence(
+            folderServices.map(_.findTestClassAndItsBuildTarget(params))
+          )
+          .flatMap { list =>
+            list.filter(_._2.nonEmpty) match {
+              case (service, buildTargets) :: Nil =>
+                service.startTestSuiteForResolved(buildTargets, params)
+              case list =>
+                if (list.isEmpty) notFound("Test class", params.testClass)
+                else
+                  definedInMultipleWorkspaceFolders(
+                    "Test class",
+                    params.testClass,
+                  )
+                Future.successful(())
+            }
+          }
+          .asJavaObject
+      case ServerCommands.StartAttach(params) if params.hostName != null =>
+        // TODO:: this should probably come with workspace name
+        val foundBuildtargets =
+          folderServices
+            .map(s => (s, s.findBuildTargetByDisplayName(params.buildTarget)))
+            .collect { case (s, Some(bt)) => (s, bt.getId()) }
+        foundBuildtargets match {
+          case (service, target) :: Nil =>
+            service.createDebugSession(target).asJavaObject
+          case list =>
+            if (list.isEmpty) notFound("Build target", params.buildTarget)
+            else
+              definedInMultipleWorkspaceFolders(
+                "Build target",
+                params.buildTarget,
+              )
+            Future.successful(()).asJavaObject
+        }
+      case ServerCommands.DiscoverAndRun(params) =>
+        getServiceFor(params.path).debugDiscovery(params)
+      case ServerCommands.AnalyzeStacktrace(content) =>
+        Future {
+          // Getting the service for focused document and first one otherwise
+          val service =
+            focusedDocument.map(getServiceFor).getOrElse(folderServices.head)
+          val command = service.analyzeStackTrace(content)
+          command.foreach(languageClient.metalsExecuteClientCommand)
+          scribe.debug(s"Executing AnalyzeStacktrace ${command}")
+        }.asJavaObject
+
+      case ServerCommands.GotoSuperMethod(textDocumentPositionParams) =>
+        getServiceFor(textDocumentPositionParams.getTextDocument().getUri())
+          .gotoSupermethod(textDocumentPositionParams)
+
+      case ServerCommands.SuperMethodHierarchy(textDocumentPositionParams) =>
+        scribe.debug(s"Executing SuperMethodHierarchy ${params.getCommand()}")
+        getServiceFor(textDocumentPositionParams.getTextDocument().getUri())
+          .superMethodHierarchy(textDocumentPositionParams)
+
+      case ServerCommands.ResetChoicePopup() =>
+        def resetPopupChoice(value: String) =
+          collectSeq(_.resetPopupChoice(value))(_ => ())
```

Comment:
This might also be good for the current folder, otherwise we could get a lot of notification about imports for example and no way to know which one belongs to which folder.